### DISCUSSION TITLE
Improving WSA launch reliability

### DISCUSSION THREAD

**Discussion by timkur** [microsoft/WSA#431] [Announcements]
We’ve been looking into reports of Windows Subsystem for Android™ (WSA) working fine for customers, and then “suddenly” one day it stops working. After this happens, WSA never starts properly again. The issue manifests itself when you attempt to launch an Android™ app, and you see the “Starting Windows Subsystem for Android™” dialog, but the app never shows up.

We’ve been able to trace this back to file corruption in one of the images we mount and have taken steps to improve our reliability in this space. Beginning with our 2307 build, fresh installs of WSA should not be impacted in this way.

Starting with the 2309 build, if we detect a failure to start based on this corruption, we will display the following dialog  

![image](https://github.com/microsoft/WSA/assets/49733346/ea6a44cc-ce2d-41da-b3e7-2269f90a89a7)

Unfortunately this means that you will lose all apps and data stored within the Windows Subsystem for Android™. This will not impact any data stored in Documents, Pictures, or any other Windows folder.

We don’t take this action lightly and are continuing to work to ensure these sorts of steps aren’t required again in the future. Thank you for using the Windows Subsystem for Android™.

**Comment by Almighty188** [comment]
There are times when it's not Data Corruption when the apps do not start but a reboot of Windows 11 will fix.  Also, as a precaution, one can always use s Windows app called WSATools which allows backing up the WSA apps and data 
so one can restore in the event of corruption after Resetting the Subsystem:
https://apps.microsoft.com/store/detail/wsatools-apk-installer-and-more/9N4P75DXL6FG?hl=en-us&gl=us&rtc=1
